### DISCUSSION TITLE
When `bootfs` is a snapshot, boot into the snapshot readonly

### DISCUSSION THREAD

**Discussion by taliaferro** [zbm-dev/zfsbootmenu#687] [Feature Requests]
Right now, if I set `bootfs=zroot/ROOT/alpine@mysnapshot`, ZFSBootMenu still shows the right filesystem (`booting into zroot/ROOT/alpine@mysnapshot`) but when it actually boots, it just boots the parent filesystem (`zroot/ROOT/alpine`) as normal.

I'd argue that if the `bootfs` is set to a snapshot, it makes more sense to try to boot into that snapshot as readonly.

**Comment by zdykstra** [comment]
It's up to the Alpine initramfs to handle `zroot/ROOT/alpine@mysnapshot` . ZBM can only verify that the snapshot is bootable (e.g. has a valid `/boot`). I don't know of any initramfs that supports booting from a read-only ZFS snapshot, and I expect that there are probably quite a few issues on every distro with supporting that use case.

**Reply by taliaferro** [reply]
Oh interesting; fair enough.
